lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -809,15 +805,9 @@ private function convertSingleBooleanValue($value, callable $callback)
             return $callback(true);
         }
 
-        /**
-         * Better safe than sorry: http://php.net/in_array#106319
-         */
-        if (in_array(strtolower(trim($value)), $this->booleanLiterals['false'], true)) {
-            return $callback(false);
-        }
-
-        if (in_array(strtolower(trim($value)), $this->booleanLiterals['true'], true)) {
-            return $callback(true);
+        $boolean = $this->booleanLiterals[strtolower(trim($value))] ?? null;
```

Comment:
Could it be just an `isset()` check? Although may be out of scope, why do we want to `trim()` the value?

Reply:
1. If we will use isset, we will need to search in array twice.
2. I don't know reason but it comes from here https://github.com/doctrine/dbal/pull/625/commits/b35b682486d76b3eea71ade1bfc4f8a4b59d0a13